src/components/App/App.js

```diff
@@ -12,36 +12,44 @@ class App extends Component {
     this.state = {
       allMovies: [],
       selectedMovie: null,
-      error: null
+      error: null,
     }
   }
 
   componentDidMount = () => {
-    getAllMovies().then(movies => {
-      this.setState({ allMovies: movies });
-    });
+    getAllMovies()
+      .then(movies => { this.setState({ allMovies: movies.movies })})
+      .catch(error => {this.setState({ error: error.toString() })});
   }
 
   chooseMovie = (movie) => {
     if (movie) {
-      getMovieById(movie.id).then(movieData => {
-        this.setState({ selectedMovie: movieData });
-      }).catch(error => {
-        this.setState({ error: error.message });
-      });
+      getMovieById(movie.id)
+        .then(movieData => {this.setState({ selectedMovie: movieData.movie })})
+        .catch(error => {this.setState({ error: error.toString() })});
     } else {
       this.setState({ selectedMovie: null });
     }
   }
 
-
   render() {
     return (
       <div className="App">
         <Header />
-        {this.state.error && <p className="error">{this.state.error}</p>}
-        {!this.state.selectedMovie && <MoviesList allMovies={this.state.allMovies} chooseMovie={this.chooseMovie} />}
-        {this.state.selectedMovie && <MovieDetails selectedMovie={this.state.selectedMovie} chooseMovie={this.chooseMovie} />}
+        {this.state.error && <p className="error">Sorry, there was an error loading your page!  {this.state.error}</p>}
+        {!this.state.selectedMovie && !this.state.error && <MoviesList allMovies={this.state.allMovies} chooseMovie={this.chooseMovie} />}
+        {this.state.selectedMovie && !this.state.error && <MovieDetails 
+          title={this.state.selectedMovie.title}
+          average_rating={this.state.selectedMovie.average_rating}
+          overview= {this.state.selectedMovie.overview}
+          genres={this.state.selectedMovie.genres}
+          budget={this.state.selectedMovie.budget}
+          revenue={this.state.selectedMovie.revenue}
+          runtime={this.state.selectedMovie.runtime}
+          tagline={this.state.selectedMovie.tagline}
+          backdrop_path={this.state.selectedMovie.backdrop_path}
+          poster_path={this.state.selectedMovie.poster_path}
+          chooseMovie={this.chooseMovie} />}
```

Comment:
I would say this is the developer's choice (and there are likely lots of opinions on this) but it would feel cleaner to me to pass the entire selectedMovie as an object then use dot notation within your MovieDetails component to access each property.  

This isn't something I would encourage you to refactor, just something to consider next time.  It's very possible you come up with an opinion differing from mine.